lib/synapse/service_watcher.rb

```diff
@@ -4,23 +4,28 @@
 module Synapse
   class ServiceWatcher
     # the method which actually dispatches watcher creation requests
-    def self.create(name, opts, synapse)
+    def self.create(name, opts, reconfigure_callback = nil, synapse)
```

Comment:
I don't see you have changed `ServiceWatcher.create` reference? Also, I think it is better to let the caller should pass in `synapse.reconfigure!` as callback explicitly.

Reply:
That's correct, I did not change the reference because I thought it it was simpler to provide a default.

Reply:
I still prefer to pass in the value explicitly if you can. The default is not `nil` but something later changed inline. Also, you don't  even pass `nil` from `ServiceWatcher.create`. Though I think the method will still work because Ruby is smart and has a few tricks up its sleeve to help determine what method arguments are being used, I would suggest NOT relying on this. I am not sure whether all versions of Ruby will be that smart. It is conventional to place any default arguments at the end of an argument list when defining a method that takes in both required and default arguments.

Reply:
Sounds good, I will do that and change the usage in the watchers